### PR TITLE
feat(payment): PI-1559 refactor Mollie in checkout-js

### PR DESCRIPTION
**Pull request by bc-ania**
## What?
Refactor the existing `Mollie` components in Checkout JS to use the new checkout payment integration JS API

## Why?
To use the new Payments API

## Testing / Proof

https://github.com/bigcommerce/checkout-js/assets/130754705/284251df-be6d-4e66-a6aa-82bfb20e6857

https://github.com/bigcommerce/checkout-js/assets/130754705/8fd6cdb9-23d6-4852-957a-6e55437be784

https://github.com/bigcommerce/checkout-js/assets/130754705/d507b4b6-0497-4f2a-9574-4533d7bbe86d

@bigcommerce/team-checkout

### CODE REVIEW

**Review by bc-peng — APPROVED** [review]
LGTM.👏

**Review by animesh1987 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -39,9 +39,6 @@ package-lock.json @bigcommerce/team-checkout @bigcommerce/team-integrations @big
 /packages/core/payment/paymentMethod/KlarnaPaymentMethod.tsx @bigcommerce/team-integrations
```
Comment (vitalii-koshovyi) [inline .github/CODEOWNERS:39]: I believe you forgot to add mollie-integrations package here

File: packages/hosted-credit-card-integration/project.json [hunk 2]
```diff
@@ -19,5 +19,5 @@
       }
     }
   },
-  "tags": ["scope:integration"]
+  "tags": ["scope:shared"]
```
Comment (vitalii-koshovyi) [inline packages/hosted-credit-card-integration/project.json:22]: should we do like this?
```suggestion
  "tags": ["scope:integration", "scope:shared"]
```

File: packages/mollie-integration/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "extends": ["../../.eslintrc.json"],
+  "overrides": [
+    {
+      "files": ["*.test.tsx", "*.tsx", "*.ts"],
```
Comment (vitalii-koshovyi) [inline packages/mollie-integration/.eslintrc.json:5]: I guess a lot of this rule should be ignored only for the `"*.test.tsx"` files

File: packages/mollie-integration/src/MollieAPMCustomForm.test.tsx [hunk 4]
```diff
@@ -59,44 +57,27 @@ describe('MollieAPMCustomForm', () => {
...
         );
 
-        expect(container.isEmptyRender()).toBe(true);
+        expect(container).toBeEmptyDOMElement();
     });
 
     it('should render MollieAPMCustomForm', () => {
-        const container = mount(<MollieAPMCustomFormTest method={method} />);
+        const { container } = render(<MollieAPMCustomFormTest method={method} />);
+        const dropdownTrigger = container.querySelector('.dropdownTrigger');
```
Comment (vitalii-koshovyi) [inline packages/mollie-integration/src/MollieAPMCustomForm.test.tsx:69]: would be better to use RTL-style queries, if it's possible. It's not a mandatory change request.

File: packages/mollie-integration/src/MolliePaymentMethod.tsx [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+
+    const instruments = checkoutState.data.getInstruments(method) || [];
+
+    const {
+        data: { getCheckout, isPaymentDataRequired },
+        statuses: { isLoadingInstruments },
+    } = checkoutState;
+
+    const checkout = getCheckout();
+    const customer = checkoutState.data.getCustomer();
```
Comment (vitalii-koshovyi) [inline packages/mollie-integration/src/MolliePaymentMethod.tsx:328]: I believe you can use it from the destructurization above

**Review by vitalii-koshovyi — COMMENTED** [review]
Great work! I have just minor change requests.

File: packages/hosted-credit-card-integration/project.json (hunk 2 above)
Comment (bc-ania) [reply packages/hosted-credit-card-integration/project.json:22]: I thought we were supposed to use only 1 tag 🤔

**Review by vitalii-koshovyi — APPROVED** [review]

**Pull request merged**